File: 0283-move-zeroes/0283-move-zeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
    let j=0;
    for (let i=0; i<nums.length; i++) { 
        if (nums[j] === 0) {
            nums.push(0);
            nums.splice(j, 1);
            j--;
        }
        j++;
    }
    return nums
};

File: 0283-move-zeroes/0283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        i, j = 0, 0
        while i + j < len(nums:
            if nums[i] == 0:
                del nums[i]
                nums.append(0)
                j += 1
            else:
                i += 1
